### DISCUSSION TITLE
Pointer Capture. Many questions.

### DISCUSSION THREAD

**Discussion by RomanSoloweow** [AvaloniaUI/Avalonia#6201] [Q&A]
In WPF, I could write Mouse.Capture anywhere. In Avalonia, this is done through an IPointer, which can only be obtained from PointerEventArgs.

Therefore, I have several questions:

1.What is the reason for this decision?
2. How to capture the mouse without the PointerEventArgs event?
3. It is recommended to use the DoubleTapped event to handle the double click event (if I'm not mistaken). But there is no reference to IPointer in the arguments of this event. Why and how to deal with such a situation?
4. What is IPointer and can it change? If I keep a reference to the IPointer object - can I do mouse capture?

**Comment by maxkatz6** [comment]
> What is the reason for this decision?

WPF input was designed with single mouse pointer in mind. UWP has way richer API here adopted to work with different pointer types (mouse, touch, pen). As a drawback, you can't have global mouse type that would cover all the cases. In Avalonia similar API was ported, although it lacks some details.

> How to capture the mouse without the PointerEventArgs event?

Subscribe and keep it, it expect it to work. But it won't help you if there is second pointer (it's common to use mouse AND touch on the same device). You should keep it in mind.

> It is recommended to use the DoubleTapped event to handle the double click event (if I'm not mistaken). But there is no reference to IPointer in the arguments of this event. Why and how to deal with such a situation?

Create an issue, it shouldn't be a problem to add it.

> What is IPointer and can it change? If I keep a reference to the IPointer object - can I do mouse capture?

I suppose it shouldn't be a problem, except the situation with multiple pointer devices. 

CC @kekekeks for some details that I could miss.

**Reply by maxkatz6** [reply]
And my question to you, why would you need Mouse.Capture globally? 
Your element can subscribe on input root pointer events (bubble or tunnel), if you will. It solves problem with multiple pointers as well. As for me it's straightforward approach.

**Reply by RomanSoloweow** [reply]
> And my question to you, why would you need Mouse.Capture globally?

For a reason unknown to me, Focus does not work for me. I call Control.Focus () but IsFocused on the element is still false.

**Reply by maxkatz6** [reply]
How is it related to the mouse capture? But in anyway, check Focusable property of the control or his children.

**Reply by kekekeks** [reply]
Note that while mouse pointer object is reused, touch pointers are ephemeral and completely disappear once the touch contact is lost. So there is no point of keeping a stale touch pointer object.

**Reply by martinrhan** [reply]
In some cases, access to PointerEventArgs is impossible. For example, when I drag the title bar, the event handlers of PointerPressed are never fired. So it is only possible to subscribe to PositionChanged of the Window. But is provides a PixelPointEventArgs which doesn't provide access to the pointer.

**Reply by IanRawley** [reply]
If Control.Focus() isn't working, make sure you're calling it on the UI thread after loading has completed.  I was having a similar problem where Focus() didn't appear to be doing anything yet, but that was because the control I was trying to focus hadn't finished loading and attaching to the visual tree.

**Reply by maxkatz6** [reply]
Yes, control needs to be attached to the visual tree and have Focusable=true set
